File: workspace/ebariaux/iPad_Console_2_0_0_Betas_ButtonEnhancement/Classes/Sensor.m
/* OpenRemote, the Home of the Digital Home.
 *  * Copyright 2008-2011, OpenRemote Inc-2009, OpenRemote Inc.
 * 
 * See the contributors.txt file in the distribution for a
 * full listing of individual contributors.
 * 
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 3.0 of
 * the License, or (at your option) any later version.
 * 
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */

#import "Sensor.h"
#import "SensorState.h"


@implementation Sensor

@synthesize sensorId, states;

- (NSString *) elementName {
	return LINK;
}

/**
 * Construct sensor instance.
 */
- (id)initWithXMLParser:(NSXMLParser *)parser elementName:(NSString *)elementName attributes:(NSDictionary *)attributeDict parentDelegate:(NSObject<NSXMLParserDelegate> *)parent {
	if (self = [super init]) {		
		sensorId = [[attributeDict objectForKey:REF] intValue];
		NSLog(@"sensor ref id=%d",sensorId);
		states = [[NSMutableArray alloc] init];
		
		xmlParserParentDelegate = [parent retain];
		[parser setDelegate:self];
	}
	return self;
}

// parse sensor state.
- (void)parser:(NSXMLParser *)parser didStartElement:(NSString *)elementName namespaceURI:(NSString *)namespaceURI qualifiedName:(NSString *)qualifiedName attributes:(NSDictionary *)attributeDict{
	if ([elementName isEqualToString:STATE]) {
		SensorState *state = [[SensorState alloc] initWithXMLParser:parser elementName:elementName attributes:attributeDict parentDelegate:self];
		[states addObject:state];
		[state release];
	} 	
}

- (void)dealloc {
	[states release];
	[super dealloc];
}

@end

File: tags/project/WebConsole/WebConsole_2_0_0/src/org/openremote/web/console/widget/Sensor.java
/* OpenRemote, the Home of the Digital Home.
* Copyright 2008-2012, OpenRemote Inc.
*
* See the contributors.txt file in the distribution for a
* full listing of individual contributors.
*
* This program is free software: you can redistribute it and/or modify
* it under the terms of the GNU Affero General Public License as
* published by the Free Software Foundation, either version 3 of the
* License, or (at your option) any later version.
*
* This program is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
* GNU Affero General Public License for more details.
*
* You should have received a copy of the GNU Affero General Public License
* along with this program. If not, see <http://www.gnu.org/licenses/>.
*/
package org.openremote.web.console.widget;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

import org.openremote.web.console.panel.entity.Link;
import org.openremote.web.console.panel.entity.StateMap;
/**
 * 
 *  
 * @author <a href="mailto:richard@openremote.org">Richard Turner</a>
 */
public class Sensor {
	Integer sensorRef;
	boolean valid = false;
	private Map<String, String> map = new LinkedHashMap<String, String>();
	
	public Sensor(Link link) {
		if (link != null) {
			this.sensorRef = link.getRef();
			List<StateMap> stateMap = link.getState();
			if (stateMap != null) {
				for (StateMap state : stateMap) {
					map.put(state.getName(), state.getValue());
				}
			}
			this.valid = true;
		}
	}
	
	public boolean isValid() {
		return valid;
	}
	
	public void setIsValid(boolean valid) {
		this.valid = valid;
	}
	
	public int getSensorRef() {
		return sensorRef;
	}
	
	public Map<String, String> getStateMap() {
		return map;
	}
	
	public String getMappedValue(String name) {
		String value = "";
		if (map != null && map.containsKey(name)) {
			value = map.get(name);
		}
		return value;
	}
}
